server/reframe/reframe.go

```diff
@@ -85,6 +85,10 @@ func (x *ReframeService) Provide(context.Context, *client.ProvideRequest) (<-cha
 
 var BitswapMetadataBytes = varint.ToUvarint(uint64(multicodec.TransportBitswap))
 
-func isBitswapMetadata(meta []byte) bool {
-	return bytes.Equal(meta, BitswapMetadataBytes)
+func containsTransportBitswap(meta []byte) bool {
+	// Metadata must be sorted according to the specification; see:
+	// - https://github.com/filecoin-project/index-provider/blob/main/metadata/metadata.go#L143
+	// This implies that if it includes Bitswap, its codec must appear at the beginning
+	// of the metadata value. Hence, bytes.HasPrefix.
+	return bytes.HasPrefix(meta, BitswapMetadataBytes)
```

Comment:
A note for future humans - this will be dependent on https://github.com/multiformats/multicodec/blob/5e275cdb00e68e76663bbcc996f2ede47bb5b301/table.csv#L138 maintaining order in the future, which is a reasonable assumption.

Reply:
:+1: good catch